src/utils/utils_windows_common.c

```diff
@@ -46,3 +46,56 @@ umf_result_t utils_duplicate_fd(int pid, int fd_in, int *fd_out) {
     (void)fd_out; // unused
     return UMF_RESULT_ERROR_NOT_SUPPORTED;
 }
+
+int util_is_symlink(const char *path) {
+    DWORD attributes = GetFileAttributesA(path);
+    if (attributes == INVALID_FILE_ATTRIBUTES) {
+        // Handle error, could not get file attributes
+        return -1;
+    }
+
+    // Check if the file is a reparse point
+    if (attributes & FILE_ATTRIBUTE_REPARSE_POINT) {
+        // It's a reparse point, which could be a symlink or a junction
+
+        // Open the file or directory with CreateFile
+        HANDLE fileHandle = CreateFileA(
+            path, 0, 0, NULL, OPEN_EXISTING,
+            FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, NULL);
+        if (fileHandle == INVALID_HANDLE_VALUE) {
+            // Handle error, could not open file
+            return -1;
+        }
+
+        // Allocate a buffer for the reparse data
+        BYTE buffer[MAXIMUM_REPARSE_DATA_BUFFER_SIZE];
+        DWORD returnedLength;
+
+        // Query the reparse data
+        BOOL result = DeviceIoControl(
+            fileHandle, FSCTL_GET_REPARSE_POINT, NULL, 0, buffer,
+            MAXIMUM_REPARSE_DATA_BUFFER_SIZE, &returnedLength, NULL);
+        if (!result) {
+            // Handle error, could not read reparse point data
+            CloseHandle(fileHandle);
```

Comment:
you can CloseHandle before if - and remove all other closeHandle in other parts of the code.